homeassistant/components/automation/state.py

```diff
@@ -42,20 +67,8 @@ def trigger(hass, config, action):
         return False
 
     if CONF_FOR in config:
```

Comment:
Better to rewrite this so that `time_delta` is always available. That way the implementation detail of `CONF_FOR in config` only has to be added here.

```python
time_delta = get_time_config(config)
if CONF_FOR in config and time_delta is None:
    return None
```